c7n/resources/sqs.py

```diff
@@ -91,7 +92,8 @@ def get_resources(self, ids, cache=True):
                 ids_normalized.append(i)
                 continue
             ids_normalized.append(i.rsplit('/', 1)[-1])
-        return super(SQS, self).get_resources(ids_normalized, cache)
+        resources = super(SQS, self).get_resources(ids_normalized, cache)
+        return [r for r in resources if Arn.parse(r['QueueArn']).resource in ids_normalized]
```

Comment:
this would otherwise pickup queues with the same prefix name, this could use a test

Reply:
the resolver test does actually verify this.